python_ta/checkers/infinite_loop_checker.py

```diff
@@ -45,6 +50,162 @@ def _check_condition_all_var_used(self, node: nodes.While) -> None:
                 node=node.test,
             )
 
+    def _check_condition_constant(self, node: nodes.While) -> None:
+        """Helper function that checks if a constant while-loop condition may lead to an infinite loop.
+
+        This helper flags loops that meet **both** of the following criteria:
+            - The `while` condition is constant (e.g., `while 1 < 2`)
+            - The loop body contains no `return`, `break`, `raise`, `yield`, or `sys.exit()` calls
+        """
+        if not self._check_constant_loop_cond(
+            node, node.test
+        ) and not self._check_constant_form_condition(node):
+            return
+        inferred = utils.safe_infer(node.test)
+        if isinstance(inferred, util.UninferableBase) or inferred is None:
+            return
+        if (
+            isinstance(inferred, nodes.Const)
+            and bool(inferred.value) is False
```

Comment:
Remembering the precedence of `and`/`or` is tricky, so I would put parentheses around each `and` clause:

```python
(isinstance(inferred, nodes.Const) and bool(inferred.value) is False)
or (... and ...)
or (... and ...)
```